theano/tensor/basic.py

```diff
@@ -147,6 +147,8 @@ def as_tensor_variable(x, name=None, ndim=None):
         If an `Apply` with more than one output is fetched.
     AsTensorError
         If `x` cannot be converted to a TensorType Variable.
+    TypeError
+        If `x` cannot be made into a Variable with `ndim` dimensions.
```

Comment:
In the params doc above, it says:
"""
ndim : None or integer
      Return a Variable with this many dimensions. Raise TypeError if it's
      not possible.
"""

I'm looking into it